tests/HL7Test.php

```diff
@@ -64,8 +64,23 @@ public function hl7_properties_can_be_set_using_chained_methods(): void
             ->withRepetitionSeparator('`')
             ->withEscapeCharacter('=')
             ->withHL7Version('555.666')
+            ->withSegmentEndingFieldSeparator(false)
             ->createMessage();
+        /**
+         * Tests:
+         *  - Field separator
+         *  - Component separator
+         *  - Subcomponent separator
+         *  - Repetition separator
+         *  - Escape character
+         */
         self::assertStringContainsString('MSH#*`=}#', $msg->toString(true));
+        /**
+         * Tests:
+         *  - Segment separator
+         *  - HL7 version
+         */
+        self::assertStringEndsWith('555.666[', $msg->toString(true));
     }
```

Comment:
Add one more test please, for `withSegmentEndingFieldSeparator(true)` and `withSegmentEndingFieldSeparator()` (considering you've set the default `true` for the parameter)

Reply:
Is this necessary? None of the other chain setting methods have that level of granularity, including testing the block for attempting to set 2+ characters as separators.

Reply:
Since users can call the method as `withSegmentEndingFieldSeparator()`, `withSegmentEndingFieldSeparator(true)` and `withSegmentEndingFieldSeparator(false)`, we need to add tests for all these scenarios. That'd protect it from breaking the signature in future